### DISCUSSION TITLE
Auto-hide

### DISCUSSION THREAD

**Discussion by varyanik** [lawand-dot-io/taskbar#83] [Feature requests]
- Add customizable value (0-5000ms) to set auto hiding
- Hide the taskbar on mouse out according to auto hide value

At the moment, the only way to hide the taskbar is by using the "Hide for Now" menu item or scroll down gesture. It's a bit annoying to do this every time, when you want to use all the screen space.

**Comment by lawand-dot-io** [comment]
Hello. Thanks for the suggestion. This feature is indeed planned to be added. Hopefully in one of the next couple of versions. Cheers

**Reply by lemoran** [reply]
Hi, will this feature be ready for v2? Or before?
This feature would be very helpful to save space when using the mac unplugged.
A cheaper way to implement this (if not auto-hiding on mouse out, which I agree is the right way) would be with a timer after selecting a task/button/window on the taskbar.

EDIT: the way to actually hide the taskbar when scrolling over (I don't use context menu "Hide for Now", it's not a very handy way) also isn't a good experience. You can hide it by scrolling up or down (a good point to have it customizable) by using the trackpad or a mouse wheel (I don't have an Apple Mouse but it works also), but if you accidentally move the pointer down to the bottom, the taskbar appears again, and this behavior feels like buggy. I think that if you implement autohide on mouse out or task change, the scrolling behavior wouldn't be needed anymore. This is my personal opinion of course.

**Reply by lawand-dot-io** [reply]
@lemoran thanks for the input and it's nice to see you're still using Taskbar. Autohide will most likely be implemented before v2. I already have an implementation but has some bugs, it was actually already included in 0.8.0 but while testing it I found it to be buggy. I will fix the bugs and release it at some point before v2 :)

**Reply by unlockweb3** [reply]
> @lemoran thanks for the input and it's nice to see you're still using Taskbar. Autohide will most likely be implemented before v2. I already have an implementation but has some bugs, it was actually already included in 0.8.0 but while testing it I found it to be buggy. I will fix the bugs and release it at some point before v2 :)

Curious, is this project still being worked on? Haven't seen any commits recently, so just wondering if I should give up hope and move on. Really like your program and am definitely willing to pay and advocate for it once v2 comes out!

**Reply by lawand-dot-io** [reply]
@derek1287 hello. Taskbar is 100% still being developed. You don't see any commits because this repo is only for bug reports and feature requests and doesn't contain any code at all. At the moment the main thing I am working on is multi-monitor support but I will also add auto hide at some point and other features too. I have a second app in development too :)

**Reply by OrkoHunter** [reply]
@lawand-dot-io I would like the auto-hide feature. How can I support you? How can I buy the paid version? Thanks!

**Reply by lawand-dot-io** [reply]
@OrkoHunter The auto-hide feature is definitely planned and will be added :) As for your offer for support it's much appreciated, the paid version will only be available after the free version expires, so just enjoy Taskbar for now, and recommend it to friends :)

**Reply by PauliusMorku** [reply]
Amazing software, would gladly pay for it, however maximizing windows and not falling behind taskbar never worked reliably. For that reason I use taskbar hiding feature and then there is no problem related to maximized windows. However, this "scroll to hide" is super annoying, it feels like the autohide feature could be implemented so easily-just move mouse away and thats it (forgive me if I'm wrong) and it would bring soooooo much value to the app. But it seems like this feature is planned for years and is not seeing the sun. Why is that?

**Reply by lawand-dot-io** [reply]
@PauliusMorku thanks for your input, one reason I keep delaying this feature is that the initial implementation never got to a point of working reliably where if you move the mouse out of the taskbar it consistently auto hides. I kept seeing an issue where you move the mouse out and the taskbar is still stuck on screen. Anyway if you're interested in testing this, I can contact you in a couple of weeks when I have some free time and create a custom build where the feature is enabled and you can try it out, otherwise I'm gonna probably work on finalizing this feature at some point early next year

**Reply by PauliusMorku** [reply]
Sure I could test it out, see if it is usable and provide you feedback.

**Reply by lawand-dot-io** [reply]
@PauliusMorku OK nice, I'll contact you in a couple of weeks
